src/components/ArchivePage/ArchivePageTemplate.js

```diff
@@ -15,95 +18,51 @@ const useStyles = makeStyles(theme => ({
+	},
+	tagDetails: {
+		margin: theme.spacing(1, 0, 0)
 	}
 }));
 
-function sortByQuarter(first, second) {
-	const firstEventName = first.parent.childYaml.name;
-	const secondEventName = second.parent.childYaml.name;
-
-	const firstEventDate = first.parent.childYaml.quarter;
-	const secondEventDate = second.parent.childYaml.quarter;
-
-	const [firstEventQuarter, firstEventYear] = firstEventDate.split(' ');
-	const [secondEventQuarter, secondEventYear] = secondEventDate.split(' ');
+function sortByQuarter(firstQuarter, secondQuarter) {
+	const [firstEventQuarter, firstEventYear] = firstQuarter.split(' ');
+	const [secondEventQuarter, secondEventYear] = secondQuarter.split(' ');
 
 	const quarterOrder = ['Winter', 'Spring', 'Summer', 'Fall'];
 
 	if (firstEventYear === secondEventYear) {
-		if (firstEventQuarter === secondEventQuarter) {
-			return firstEventName < secondEventName ? -1 : 1;
-		}
 		return quarterOrder.indexOf(secondEventQuarter) - quarterOrder.indexOf(firstEventQuarter);
 	}
 	return parseInt(secondEventYear) - parseInt(firstEventYear);
 }
 
-function getQuarterList(events) {
-	const sortedQuarters = [];
-	events.forEach(event => {
-		const { quarter } = event.parent.childYaml;
-		if (!sortedQuarters.includes(quarter)) {
-			sortedQuarters.push(quarter);
-		}
-	});
-	return sortedQuarters;
-}
-
-function getQuarterEvents(events) {
-	const quarterEvents = {};
-	events.forEach(event => {
-		const { quarter } = event.parent.childYaml;
-		if (quarterEvents[quarter] === undefined) {
-			quarterEvents[quarter] = [];
-		}
-		quarterEvents[quarter].push(event.parent.childYaml);
-	});
-	return quarterEvents;
+function getQuarterList(quarterEvents) {
+	const quarterList = [];
+	for (const property in quarterEvents) {
+		quarterList.push(property);
+	}
+	quarterList.sort(sortByQuarter);
+	return quarterList;
 }
 
-function ArchivePageTemplate() {
+function ArchivePageTemplate({ pageContext }) {
 	const classes = useStyles();
-	const pageQuery = useStaticQuery(graphql`
-		query WorkshopArchiveQuery {
-			allYaml(sort: {fields: quarter, order: DESC}) {
-				nodes {
-					parent {
-						... on File {
-							id
-							childYaml {
-								director
-								name
-								mainLink
-								quarter
-								tags
-								workshops {
-									name
-									repo
-									slides
-									tags
-									youtube
-									presenter
-								}
-							}
-						}
-					}
-				}
-			}
-		}
-	`);
-	const events = pageQuery.allYaml.nodes;
-	events.sort(sortByQuarter);
-	const sortedQuarterList = getQuarterList(events);
-	const quarterEventsDict = getQuarterEvents(events);
-
-	const quarterEvents = sortedQuarterList.map(quarter =>
+	const { quarterEvents, allTags } = pageContext;
+	const quarterList = getQuarterList(quarterEvents);
+	for (const event in quarterEvents) {
+		quarterEvents[event].sort((first, second) => first.name < second.name ? -1 : 1);
+	}
+	allTags.sort((first, second) => first.displayName < second.displayName ? -1 : 1);
```

Comment:
I don't think it matters much, but technically the sorting function isn't right, since it is not reflexive. And `.sort` [requires](https://tc39.es/ecma262/#_ref_9097) reflexivity to function properly.

To get out of the spec correctness purgatory, I'd just make a helper function `cmp` that does the right thing:

```js
function cmp(a, b) {
  if (a === b) {
    return 0;
  } else if (a < b) {
    return -1;
  } else {
    return 1;
  }
}
allTags.sort((first, second) => cmp(first.displayName, second.displayName));
```